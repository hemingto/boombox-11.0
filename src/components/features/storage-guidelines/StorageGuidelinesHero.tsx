/**
 * @fileoverview Storage guidelines page hero section component
 * @source boombox-10.0/src/app/components/storage-guidelines/storageguidelineshero.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays the hero section for the storage guidelines page with heading and subtitle.
 * Uses semantic HTML and responsive padding following design system standards.
 * 
 * API ROUTES UPDATED:
 * - None (presentational component only)
 * 
 * DESIGN SYSTEM UPDATES:
 * - Uses existing responsive padding patterns (lg:px-16 px-6)
 * - Maintains semantic spacing with my-10 sm:my-20
 * - Semantic h1 and p elements for proper content hierarchy
 * 
 * @refactor Renamed file to PascalCase, added comprehensive documentation
 */

'use client';

import React from 'react';

/**
 * StorageGuidelinesHero component
 * 
 * Simple hero section for the storage guidelines page displaying the page title
 * and a subtitle describing the content (best practices).
 * Uses responsive padding and spacing following the design system.
 * 
 * @example
 * ```tsx
 * <StorageGuidelinesHero />
 * ```
 */
export function StorageGuidelinesHero(): React.ReactElement {
  return (
    <div className="w-full my-10 sm:my-20 lg:px-16 px-6">
      <div>
        <h1 className="mb-4">Storage Guidelines</h1>
        <p>best practices</p>
      </div>
    </div>
  );
}

