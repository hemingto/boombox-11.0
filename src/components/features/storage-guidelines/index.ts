/**
 * @fileoverview Storage guidelines feature components exports
 * @source boombox-10.0/src/app/components/storage-guidelines/
 */

export { StorageGuidelinesHero } from './StorageGuidelinesHero';
export { StorageGuidelinesList } from './StorageGuidelinesList';

