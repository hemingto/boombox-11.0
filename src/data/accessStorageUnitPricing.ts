export const accessStorageUnitPricing = 45;

