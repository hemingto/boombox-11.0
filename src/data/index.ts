/**
 * @fileoverview Data exports for boombox-11.0
 * @refactor Centralized data exports for all static/configuration data
 */

// Customer Reviews
export * from './customerReviews';

// FAQ Data
export * from './faq';

// Insurance Options
export * from './insuranceOptions';

// Labor Options
export * from './laborOptions';

// Locations
export * from './locations';

// Bay Area Cities
export * from './bayareacities';

// Bay Area Service Area Coordinates
export * from './bayareaserviceareacoordinates';

// Popular Locations
export * from './popularlocations';

// Training Videos
export * from './trainingVideos';

// Sitemap Links
export * from './sitemapLinks';

// Storage Guidelines (contains JSX, uses .tsx extension)
export * from './storageGuidelines';

// Pricing Data
export * from './accessStorageUnitPricing';
export * from './cancelationpricing';
export * from './zipcodeprices';

