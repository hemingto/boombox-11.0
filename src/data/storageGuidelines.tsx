/**
 * @fileoverview Storage guidelines data for the storage guidelines page
 * @source boombox-10.0/src/app/components/storage-guidelines/storageguidelineslist.tsx (inline data)
 * @refactor Extracted storage guidelines data to centralized data folder for better maintainability
 */

import React from 'react';

/**
 * Storage guideline interface
 */
export interface StorageGuideline {
  number: number;
  title: string;
  description: React.ReactNode;
}

/**
 * Storage guidelines data for Boombox storage service
 * 
 * Contains all rules and best practices for using Boombox storage containers:
 * - Minimum storage terms
 * - Size and packing requirements
 * - Safety restrictions
 * - Driver responsibilities
 */
export const storageGuidelines: StorageGuideline[] = [
  {
    number: 1,
    title: 'Storage terms are a minimum of 2 months',
    description: (
      <p>All storage terms are a minimum of two months. You can access your items before the 2 month term, however you will be charged for a full two months of storage.</p>
    ),
  },
  {
    number: 2,
    title: 'All items must fit into your Boombox storage container',
    description: (
      <>
        <p className="mb-2">All items need to fit into our standard 5x8 Boombox storage container. The door to your unit must be able to close properly. If the door is unable to be properly closed you must remove the stored item that is preventing the door from closing.</p>
        <p>If you are unsure if one of your storage items will fit please reach out at our Help Center.</p>
      </>
    ),
  },
  {
    number: 3,
    title: 'Pack Smart',
    description: (
      <>
        <p className="mb-4">We&apos;ll always handle your stuff with care, but you&apos;re responsible for ensuring your Boombox is packed properly. Please keep in mind the following to ensure your Boombox is properly packed:</p>
        <p className="mb-2"><span className="font-bold">1. Organize Your Items into Tiers:</span> Think of each tier as a wall or stack within the container, starting with the heaviest items at the bottom and lighter items on top. This method helps maintain balance and stability during transit.</p>
        <p className="mb-2"><span className="font-bold">2. Strategize Your Loading Process:</span> Begin your loading at the back of the container and move forward. The base of each tier, composed of your sturdiest items like dressers or desks, should support the weight of the items above.</p>
        <p className="mb-2"><span className="font-bold">3. Distribute Weight Evenly:</span> Place heavy items around the base of the container to evenly distribute weight, ensuring a stable and secure load.</p>
        <p className="mb-4"><span className="font-bold">4. Fill Gaps and Protect Items:</span> Utilize soft materials like furniture pads, blankets, and pillows to fill spaces between items, preventing shifts and scratches. Disassemble furniture to save space and wrap each piece to protect from damage.</p>
        <p className="mb-2">Although not required, it is best practice that you disassemble furniture items (bed frames, tables) before loading them into your Boombox container. You&apos;ll ensure your items are transported safely while also freeing up space to pack more items.</p>
        <p className="mb-2">If you plan on storing a mattress, we strongly recommend you cover it in a mattress cover. We can take mattresses without a cover, but it is best practice to keep mattresses covered to prevent damage over time.</p>
      </>
    ),
  },
  {
    number: 4,
    title: 'Inform us about heavy items or difficult moving conditions',
    description: (
      <>
        <p className="mb-2">If you&apos;re storing items heavier than 50 pounds, give your moving crew a heads up to ensure they&apos;re ready for your larger pieces. For anything over 150lbs, please inform us. Each Boombox container can hold up to 2000lbs, and we can&apos;t transport containers exceeding this weight.</p>
        <p>Also, if you&apos;re moving from a hard-to-reach spot (like a 10th floor apartment), or if you&apos;re in a busy area with limited parking, let us know in advance so we can plan accordingly.</p>
      </>
    ),
  },
  {
    number: 5,
    title: "Don't store anything that's illegal and/or unsafe",
    description: (
      <>
        <p className="mb-2">The safety of our employees, our customers, and all of their belongings always comes first. The following items are banned from being stored in Boombox:</p>
        <p className="mb-2">Anything that goes boom, such as guns (or weapons of any kind), explosives, fireworks, and other flammables like gas, oil, kerosene, paint, and lighter fluid. Anything that is or was alive. This includes things like fruit, meats, cheeses, animals, insects, fungal or bacterial cultures, etc. Anything that is (or was) edible. This includes any perishable and non-perishable food items. Anything that&apos;s illegal, such as drugs, drug paraphernalia, stolen property, and anything else that you can get arrested for possessing. Anything that smells, oozes, leaks, or bursts, such as hazardous items, toxic materials, items that produce gas or odors, any container with liquids, items that produce loud or disruptive noises, and items that may increase in size or burst. Basically, anything that might get outside of your box or bin and harm others, our customers&apos; and employees&apos; belongings, or our storage facility.</p>
      </>
    ),
  },
  {
    number: 6,
    title: 'Your Boombox Driver is not allowed to help load your storage unit',
    description: (
      <>
        <p className="mb-2">Although we want to be as helpful as possible, Boombox drivers are not insured or licensed to help load your items into your storage unit.</p>
        <p className="mb-2">They should not be expected help load your items and are there only to transport your container once your unit is loaded. They&apos;ll make sure your unit is parked in a safe space for loading and give you access to moving supplies, however, they aren&apos;t allowed to assist in the loading process.</p>
        <p className="mb-2">If a Boombox Driver, assists with loading your items no damages to your items or building are covered under our insurance coverage.</p>
        <p>If you need moving help please add it to your order and you&apos;ll be matched with a local pro.</p>
      </>
    ),
  },
];

