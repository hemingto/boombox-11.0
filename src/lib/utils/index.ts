/**
 * @fileoverview Central utility exports for boombox-11.0
 * Organized by domain and functionality for clean imports
 */

// Core utility functions
export * from './cn';
export * from './phoneUtils';
export * from './currencyUtils';
export * from './navigationUtils';
export * from './emailUtils';

// Notification utilities
export * from './notificationUtils';
export * from './validationUtils';
export * from './formatUtils';
export * from './statusUtils';

// Blog utilities
export * from './blogUtils';

// Explicit exports to resolve conflicts
export {
  calculateMonthlyStorageCost,
  isInServiceArea,
  getServiceAreaForZip,
} from './businessUtils';

export {
  formatDateForInput,
  formatDateForDisplay,
  formatDateCompact,
  formatDateTime,
  formatTime,
  formatTime24Hour,
  formatTimestamp,
  parseAppointmentTime,
  calculateDeliveryWindow,
  getUnitSpecificStartTime,
  formatVerboseDate,
} from './dateUtils';

export {
  generateDriverToken,
  generateCustomerMoverChangeToken,
  notifyDriverReassignment,
  notifyCustomerMoverChange,
  recordDriverCancellation,
  recordMoverCancellation,
  type DriverCancellationData,
  type MoverCancellationData,
  type AvailableDriver,
} from './cancellationUtils';

// Domain-specific utilities (avoiding conflicts)
// Note: appointmentUtils contains parseLoadingHelpPrice which conflicts with pricingUtils
// Using pricingUtils version for parseLoadingHelpPrice
export {
  formatAppointmentTime,
  formatAppointmentDateForSms,
  formatTimeMinusOneHour,
  validateAppointmentDateTime,
  calculateAppointmentChanges,
  calculateFinalUnitCount,
  getUnitNumbersToRemove,
  getDayOfWeekForAvailability,
  validateMovingPartnerAvailability,
  generateDriverReconfirmToken,
  generateDriverWebViewUrl,
  formatDriverInfo,
  verifyTrackingToken,
  // parseLoadingHelpPrice - using pricingUtils version
} from './appointmentUtils';

// Explicit exports from availabilityUtils to avoid conflicts
export {
  DEFAULT_BUSINESS_HOURS,
  calculateDriverRequirements,
  formatTimeLocal,
  generateBusinessHourSlots,
  hasTimeConflict,
  isResourceAvailableInSlot,
  createTimeConflict,
  checkOnfleetTaskConflicts,
  determineAvailabilityLevel,
  getDistinctDaysOfWeekInMonth,
  groupBy,
  isDateInPast,
  generateCacheKey,
  getMinutesDifference,
  isValidTimeSlot,
  timeStringToMinutes,
  // getDayOfWeekString - conflicts with dateFormattingUtils, using that version
} from './availabilityUtils';

export * from './moverChangeUtils';

// Explicit exports from movingPartnerUtils to avoid conflicts
export {
  createDefaultMoverAvailability,
  parseTimeToMinutes,
  checkMoverExists,
  createMover,
  findAvailableMovingPartners,
  uploadFileToCloudinary,
  deleteOldCloudinaryFile,
  getMovingPartnerJobs,
  assignMovingPartnerDriver,
  // getDayOfWeekString - conflicts with other modules, already exported from dateFormattingUtils
} from './movingPartnerUtils';

// Messaging and communication utilities
// Note: twilioUtils contains server-side code (headers) - import directly in API routes only
export * from './messageClassificationUtils';
export * from './inboundMessageUtils';
// Note: dateFormattingUtils contains formatAppointmentTime which conflicts with appointmentUtils
// Using appointmentUtils version for appointment-specific formatting
// export * from './dateFormattingUtils';
// Explicit export of non-conflicting functions from dateFormattingUtils
export { formatAppointmentDate } from './dateFormattingUtils';

// Driver utilities (explicit exports to avoid conflicts)
export {
  getDriverTeamIds,
  createDefaultDriverAvailability,
  formatPhoneForOnfleet,
  mapVehicleTypeToOnfleet,
  buildOnfleetWorkerPayload,
  validateDriverUniqueness,
  findDriverInvitation,
  validateInvitationStatus,
  DEFAULT_DRIVER_SERVICES,
  DEFAULT_MOVING_PARTNER_VEHICLE_TYPE,
  DEFAULT_DRIVER_CAPACITY,
  DRIVER_STATUSES,
  type DriverRegistrationData,
  type OnfleetWorkerPayload,
  type DriverApprovalResult,
} from './driverUtils';

// Pricing utilities (explicit exports to avoid conflicts)
export {
  getBoomboxPriceByZipCode,
  formatStorageUnitPrice,
  formatInsurancePrice,
  parseLoadingHelpPrice, // Using pricingUtils version as canonical
} from './pricingUtils';

// Storage utilities (explicit exports)
export {
  getStorageUnitText,
} from './storageUtils';

// Sorting and pagination utilities
export * from './sortingUtils';

// Tracking utilities
export * from './trackingStatusUtils';
