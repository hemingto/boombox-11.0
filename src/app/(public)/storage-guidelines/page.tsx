/**
 * @fileoverview Storage guidelines page - what can/cannot be stored
 * @source boombox-10.0/src/app/storage-guidelines/page.tsx
 * @refactor Migrated to (public) route group with SEO metadata
 */

'use client';

import type { Metadata } from 'next';
import {
  StorageGuidelinesHero,
  StorageGuidelinesList,
} from '@/components/features/storage-guidelines';
import { FaqSection, HelpCenterSection } from '@/components/features/landing';

// Note: Page is client component
// SEO metadata should be added via generateMetadata when converting to server component

export default function StorageGuidelines() {
  return (
    <>
      <StorageGuidelinesHero />
      <StorageGuidelinesList />
      <FaqSection />
      <HelpCenterSection />
    </>
  );
}

