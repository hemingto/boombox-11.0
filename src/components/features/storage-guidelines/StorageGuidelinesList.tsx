/**
 * @fileoverview Storage guidelines list component displaying numbered guidelines with descriptions
 * @source boombox-10.0/src/app/components/storage-guidelines/storageguidelineslist.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays all storage guidelines as numbered cards with titles and detailed descriptions.
 * Each guideline includes rules, best practices, and safety information for using Boombox storage.
 * Data extracted to @/data/storageGuidelines for better maintainability.
 * 
 * API ROUTES UPDATED:
 * - None (presentational component only)
 * 
 * DESIGN SYSTEM UPDATES:
 * - border-slate-100 → border-border (semantic border color)
 * - border-zinc-950 → border-primary (semantic primary color for numbered badges)
 * - text-zinc-950 → text-primary (semantic text color for badges)
 * - Maintains responsive padding (lg:px-16 px-6)
 * - Preserves spacing and layout patterns
 * 
 * @refactor Extracted data to centralized data folder, renamed to PascalCase, applied design system colors,
 * removed hardcoded Help Center link (will be handled by parent component)
 */

import React from 'react';
import Link from 'next/link';
import { storageGuidelines } from '@/data/storageGuidelines';
import type { StorageGuideline } from '@/data/storageGuidelines';

/**
 * StorageGuidelinesList component
 * 
 * Displays a list of storage guidelines as numbered cards.
 * Each guideline includes a number badge, title, and detailed description.
 * The descriptions may contain multiple paragraphs and inline links.
 * 
 * @example
 * ```tsx
 * <StorageGuidelinesList />
 * ```
 */
export function StorageGuidelinesList(): React.ReactElement {
  return (
    <div className="lg:px-16 px-6">
      {storageGuidelines.map((guideline: StorageGuideline) => (
        <article 
          key={guideline.number} 
          className="flex mb-8 p-6 border border-border rounded-md items-start"
          aria-labelledby={`guideline-${guideline.number}-title`}
        >
          {/* Number Badge */}
          <div 
            className="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
            aria-label={`Guideline ${guideline.number}`}
          >
            {guideline.number}
          </div>
          
          {/* Content */}
          <div>
            <h2 id={`guideline-${guideline.number}-title`} className="mb-5">
              {guideline.title}
            </h2>
            <div className="text-text-primary">
              {/* Special handling for guideline 2 to add Help Center link */}
              {guideline.number === 2 ? (
                <>
                  <p className="mb-2">All items need to fit into our standard 5x8 Boombox storage container. The door to your unit must be able to close properly. If the door is unable to be properly closed you must remove the stored item that is preventing the door from closing.</p>
                  <p>If you are unsure if one of your storage items will fit please reach out at our <Link href="/help-center" className="font-bold underline">Help Center</Link>.</p>
                </>
              ) : (
                guideline.description
              )}
            </div>
          </div>
        </article>
      ))}
    </div>
  );
}

