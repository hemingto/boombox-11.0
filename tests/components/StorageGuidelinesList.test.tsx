/**
 * @fileoverview Tests for StorageGuidelinesList component
 * Following boombox-11.0 testing standards (97â†’0 failure learnings)
 */
import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { testAccessibility } from '../utils/accessibility';
import { StorageGuidelinesList } from '@/components/features/storage-guidelines/StorageGuidelinesList';
import { storageGuidelines } from '@/data/storageGuidelines';

expect.extend(toHaveNoViolations);

describe('StorageGuidelinesList', () => {
  // REQUIRED: Test cleanup
  beforeEach(() => {
    jest.clearAllMocks();
  });

  // REQUIRED: Basic rendering
  describe('Rendering', () => {
    it('renders without crashing', () => {
      render(<StorageGuidelinesList />);
      expect(screen.getByRole('heading', { level: 2, name: /Storage terms are a minimum of 2 months/i })).toBeInTheDocument();
    });

    it('renders all storage guidelines', () => {
      render(<StorageGuidelinesList />);
      
      // Should render all guidelines from data
      const articles = screen.getAllByRole('article');
      expect(articles).toHaveLength(storageGuidelines.length);
    });

    it('renders guideline titles correctly', () => {
      render(<StorageGuidelinesList />);
      
      storageGuidelines.forEach((guideline) => {
        const heading = screen.getByRole('heading', { level: 2, name: guideline.title });
        expect(heading).toBeInTheDocument();
      });
    });

    it('renders number badges for each guideline', () => {
      render(<StorageGuidelinesList />);
      
      storageGuidelines.forEach((guideline) => {
        const badge = screen.getByLabelText(`Guideline ${guideline.number}`);
        expect(badge).toBeInTheDocument();
        expect(badge).toHaveTextContent(String(guideline.number));
      });
    });

    it('renders Help Center link in guideline 2', () => {
      render(<StorageGuidelinesList />);
      
      const helpCenterLink = screen.getByRole('link', { name: /help center/i });
      expect(helpCenterLink).toBeInTheDocument();
      expect(helpCenterLink).toHaveAttribute('href', '/help-center');
    });
  });

  // MANDATORY: Accessibility testing
  describe('Accessibility', () => {
    it('has no accessibility violations', async () => {
      const renderResult = render(<StorageGuidelinesList />);
      await testAccessibility(renderResult);
    });

    it('uses semantic HTML with article elements', () => {
      render(<StorageGuidelinesList />);
      
      const articles = screen.getAllByRole('article');
      expect(articles.length).toBeGreaterThan(0);
    });

    it('has proper heading hierarchy', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      // All guideline titles should be h2
      const h2Headings = container.querySelectorAll('h2');
      expect(h2Headings.length).toBe(storageGuidelines.length);
    });

    it('articles have proper aria-labelledby', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const articles = container.querySelectorAll('article');
      articles.forEach((article, index) => {
        const guidelineNumber = index + 1;
        expect(article).toHaveAttribute('aria-labelledby', `guideline-${guidelineNumber}-title`);
      });
    });

    it('headings have proper id attributes', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const headings = container.querySelectorAll('h2');
      headings.forEach((heading, index) => {
        const guidelineNumber = index + 1;
        expect(heading).toHaveAttribute('id', `guideline-${guidelineNumber}-title`);
      });
    });

    it('number badges have descriptive aria-labels', () => {
      render(<StorageGuidelinesList />);
      
      storageGuidelines.forEach((guideline) => {
        const badge = screen.getByLabelText(`Guideline ${guideline.number}`);
        expect(badge).toHaveAttribute('aria-label', `Guideline ${guideline.number}`);
      });
    });

    it('Help Center link is accessible', () => {
      render(<StorageGuidelinesList />);
      
      const link = screen.getByRole('link', { name: /help center/i });
      expect(link).toHaveClass('underline');
      expect(link).toHaveAttribute('href', '/help-center');
    });
  });

  // Design System Integration
  describe('Design System Compliance', () => {
    it('applies responsive padding classes to container', () => {
      const { container } = render(<StorageGuidelinesList />);
      const outerDiv = container.firstChild as HTMLElement;
      
      expect(outerDiv).toHaveClass('lg:px-16');
      expect(outerDiv).toHaveClass('px-6');
    });

    it('uses semantic border colors for guideline cards', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const articles = container.querySelectorAll('article');
      articles.forEach((article) => {
        expect(article).toHaveClass('border-border');
      });
    });

    it('uses semantic primary color for number badges', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const badges = container.querySelectorAll('[aria-label^="Guideline"]');
      badges.forEach((badge) => {
        expect(badge).toHaveClass('border-primary');
        expect(badge).toHaveClass('text-primary');
      });
    });

    it('applies consistent spacing between guidelines', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const articles = container.querySelectorAll('article');
      articles.forEach((article) => {
        expect(article).toHaveClass('mb-8');
      });
    });
  });

  // Card Styling
  describe('Card Styling', () => {
    it('applies correct styling to guideline cards', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const articles = container.querySelectorAll('article');
      articles.forEach((article) => {
        expect(article).toHaveClass('flex');
        expect(article).toHaveClass('p-6');
        expect(article).toHaveClass('border');
        expect(article).toHaveClass('rounded-md');
        expect(article).toHaveClass('items-start');
      });
    });

    it('applies correct styling to number badges', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const badges = container.querySelectorAll('[aria-label^="Guideline"]');
      badges.forEach((badge) => {
        expect(badge).toHaveClass('flex');
        expect(badge).toHaveClass('items-center');
        expect(badge).toHaveClass('justify-center');
        expect(badge).toHaveClass('w-8');
        expect(badge).toHaveClass('h-8');
        expect(badge).toHaveClass('rounded-full');
        expect(badge).toHaveClass('border-2');
        expect(badge).toHaveClass('font-semibold');
      });
    });

    it('applies margin to headings', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const headings = container.querySelectorAll('h2');
      headings.forEach((heading) => {
        expect(heading).toHaveClass('mb-5');
      });
    });
  });

  // Data Integration
  describe('Data Integration', () => {
    it('renders data from imported storageGuidelines', () => {
      render(<StorageGuidelinesList />);
      
      // Verify first guideline from data
      const firstGuideline = storageGuidelines[0];
      expect(screen.getByRole('heading', { level: 2, name: firstGuideline.title })).toBeInTheDocument();
    });

    it('renders all guideline numbers sequentially', () => {
      render(<StorageGuidelinesList />);
      
      storageGuidelines.forEach((guideline) => {
        const badge = screen.getByLabelText(`Guideline ${guideline.number}`);
        expect(badge).toHaveTextContent(String(guideline.number));
      });
    });

    it('renders correct number of guidelines', () => {
      render(<StorageGuidelinesList />);
      
      const articles = screen.getAllByRole('article');
      expect(articles).toHaveLength(6); // Based on data
    });
  });

  // Component Structure
  describe('Component Structure', () => {
    it('uses flexbox layout for each guideline card', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const articles = container.querySelectorAll('article');
      articles.forEach((article) => {
        expect(article).toHaveClass('flex');
      });
    });

    it('number badges are flex-shrink-0', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const badges = container.querySelectorAll('[aria-label^="Guideline"]');
      badges.forEach((badge) => {
        expect(badge).toHaveClass('flex-shrink-0');
      });
    });

    it('badges have proper spacing from content', () => {
      const { container } = render(<StorageGuidelinesList />);
      
      const badges = container.querySelectorAll('[aria-label^="Guideline"]');
      badges.forEach((badge) => {
        expect(badge).toHaveClass('mr-4');
        expect(badge).toHaveClass('mt-0.5');
      });
    });
  });

  // Content Verification
  describe('Content', () => {
    it('renders first guideline correctly', () => {
      render(<StorageGuidelinesList />);
      
      expect(screen.getByRole('heading', { name: /Storage terms are a minimum of 2 months/i })).toBeInTheDocument();
      expect(screen.getByText(/All storage terms are a minimum of two months/i)).toBeInTheDocument();
    });

    it('renders last guideline correctly', () => {
      render(<StorageGuidelinesList />);
      
      expect(screen.getByRole('heading', { name: /Your Boombox Driver is not allowed to help load/i })).toBeInTheDocument();
      expect(screen.getByText(/Boombox drivers are not insured or licensed/i)).toBeInTheDocument();
    });

    it('includes special handling for guideline 2 with link', () => {
      render(<StorageGuidelinesList />);
      
      // Guideline 2 should have Help Center link
      const guideline2 = screen.getByRole('article', { 
        name: 'All items must fit into your Boombox storage container' 
      });
      expect(guideline2).toBeInTheDocument();
      
      const helpCenterLink = screen.getByRole('link', { name: /help center/i });
      expect(helpCenterLink).toBeInTheDocument();
    });
  });

  // Snapshot Testing
  describe('Snapshot', () => {
    it('matches snapshot', () => {
      const { container } = render(<StorageGuidelinesList />);
      expect(container).toMatchSnapshot();
    });
  });
});

