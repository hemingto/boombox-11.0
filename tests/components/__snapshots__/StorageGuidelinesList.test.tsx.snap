// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StorageGuidelinesList Snapshot matches snapshot 1`] = `
<div>
  <div
    class="lg:px-16 px-6"
  >
    <article
      aria-labelledby="guideline-1-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 1"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        1
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-1-title"
        >
          Storage terms are a minimum of 2 months
        </h2>
        <div
          class="text-text-primary"
        >
          <p>
            All storage terms are a minimum of two months. You can access your items before the 2 month term, however you will be charged for a full two months of storage.
          </p>
        </div>
      </div>
    </article>
    <article
      aria-labelledby="guideline-2-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 2"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        2
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-2-title"
        >
          All items must fit into your Boombox storage container
        </h2>
        <div
          class="text-text-primary"
        >
          <p
            class="mb-2"
          >
            All items need to fit into our standard 5x8 Boombox storage container. The door to your unit must be able to close properly. If the door is unable to be properly closed you must remove the stored item that is preventing the door from closing.
          </p>
          <p>
            If you are unsure if one of your storage items will fit please reach out at our 
            <a
              class="font-bold underline"
              href="/help-center"
            >
              Help Center
            </a>
            .
          </p>
        </div>
      </div>
    </article>
    <article
      aria-labelledby="guideline-3-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 3"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        3
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-3-title"
        >
          Pack Smart
        </h2>
        <div
          class="text-text-primary"
        >
          <p
            class="mb-4"
          >
            We'll always handle your stuff with care, but you're responsible for ensuring your Boombox is packed properly. Please keep in mind the following to ensure your Boombox is properly packed:
          </p>
          <p
            class="mb-2"
          >
            <span
              class="font-bold"
            >
              1. Organize Your Items into Tiers:
            </span>
             Think of each tier as a wall or stack within the container, starting with the heaviest items at the bottom and lighter items on top. This method helps maintain balance and stability during transit.
          </p>
          <p
            class="mb-2"
          >
            <span
              class="font-bold"
            >
              2. Strategize Your Loading Process:
            </span>
             Begin your loading at the back of the container and move forward. The base of each tier, composed of your sturdiest items like dressers or desks, should support the weight of the items above.
          </p>
          <p
            class="mb-2"
          >
            <span
              class="font-bold"
            >
              3. Distribute Weight Evenly:
            </span>
             Place heavy items around the base of the container to evenly distribute weight, ensuring a stable and secure load.
          </p>
          <p
            class="mb-4"
          >
            <span
              class="font-bold"
            >
              4. Fill Gaps and Protect Items:
            </span>
             Utilize soft materials like furniture pads, blankets, and pillows to fill spaces between items, preventing shifts and scratches. Disassemble furniture to save space and wrap each piece to protect from damage.
          </p>
          <p
            class="mb-2"
          >
            Although not required, it is best practice that you disassemble furniture items (bed frames, tables) before loading them into your Boombox container. You'll ensure your items are transported safely while also freeing up space to pack more items.
          </p>
          <p
            class="mb-2"
          >
            If you plan on storing a mattress, we strongly recommend you cover it in a mattress cover. We can take mattresses without a cover, but it is best practice to keep mattresses covered to prevent damage over time.
          </p>
        </div>
      </div>
    </article>
    <article
      aria-labelledby="guideline-4-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 4"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        4
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-4-title"
        >
          Inform us about heavy items or difficult moving conditions
        </h2>
        <div
          class="text-text-primary"
        >
          <p
            class="mb-2"
          >
            If you're storing items heavier than 50 pounds, give your moving crew a heads up to ensure they're ready for your larger pieces. For anything over 150lbs, please inform us. Each Boombox container can hold up to 2000lbs, and we can't transport containers exceeding this weight.
          </p>
          <p>
            Also, if you're moving from a hard-to-reach spot (like a 10th floor apartment), or if you're in a busy area with limited parking, let us know in advance so we can plan accordingly.
          </p>
        </div>
      </div>
    </article>
    <article
      aria-labelledby="guideline-5-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 5"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        5
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-5-title"
        >
          Don't store anything that's illegal and/or unsafe
        </h2>
        <div
          class="text-text-primary"
        >
          <p
            class="mb-2"
          >
            The safety of our employees, our customers, and all of their belongings always comes first. The following items are banned from being stored in Boombox:
          </p>
          <p
            class="mb-2"
          >
            Anything that goes boom, such as guns (or weapons of any kind), explosives, fireworks, and other flammables like gas, oil, kerosene, paint, and lighter fluid. Anything that is or was alive. This includes things like fruit, meats, cheeses, animals, insects, fungal or bacterial cultures, etc. Anything that is (or was) edible. This includes any perishable and non-perishable food items. Anything that's illegal, such as drugs, drug paraphernalia, stolen property, and anything else that you can get arrested for possessing. Anything that smells, oozes, leaks, or bursts, such as hazardous items, toxic materials, items that produce gas or odors, any container with liquids, items that produce loud or disruptive noises, and items that may increase in size or burst. Basically, anything that might get outside of your box or bin and harm others, our customers' and employees' belongings, or our storage facility.
          </p>
        </div>
      </div>
    </article>
    <article
      aria-labelledby="guideline-6-title"
      class="flex mb-8 p-6 border border-border rounded-md items-start"
    >
      <div
        aria-label="Guideline 6"
        class="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
      >
        6
      </div>
      <div>
        <h2
          class="mb-5"
          id="guideline-6-title"
        >
          Your Boombox Driver is not allowed to help load your storage unit
        </h2>
        <div
          class="text-text-primary"
        >
          <p
            class="mb-2"
          >
            Although we want to be as helpful as possible, Boombox drivers are not insured or licensed to help load your items into your storage unit.
          </p>
          <p
            class="mb-2"
          >
            They should not be expected help load your items and are there only to transport your container once your unit is loaded. They'll make sure your unit is parked in a safe space for loading and give you access to moving supplies, however, they aren't allowed to assist in the loading process.
          </p>
          <p
            class="mb-2"
          >
            If a Boombox Driver, assists with loading your items no damages to your items or building are covered under our insurance coverage.
          </p>
          <p>
            If you need moving help please add it to your order and you'll be matched with a local pro.
          </p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
